src/parser/xsd/element.js

```diff
@@ -41,9 +41,6 @@ class Element extends XSDElement {
       var refDescriptor = this.ref.describe(definitions);
       if (refDescriptor) {
         this.descriptor = descriptor = refDescriptor.clone(isMany);
-        if (this.$nillable) {
```

Comment:
This is redundant. It would already be run by the same code above, which is not nested inside of another conditional.